spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/LogisticRegressionClassifier.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.spark;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.StageConfigurer;
+import co.cask.cdap.etl.api.batch.SparkCompute;
+import co.cask.cdap.etl.api.batch.SparkExecutionPluginContext;
+import com.google.common.base.Preconditions;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.classification.LogisticRegressionModel;
+import org.apache.spark.mllib.feature.HashingTF;
+import org.apache.twill.filesystem.Location;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * SparkCompute that uses a trained Logistic Regression model to classify and tag input records.
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name(LogisticRegressionClassifier.PLUGIN_NAME)
+@Description("Uses a Logistic Regression model to classify records.")
+public class LogisticRegressionClassifier extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+  public static final String PLUGIN_NAME = "LogisticRegressionClassifier";
+  private Config config;
+  private Schema outputSchema;
+
+  private LogisticRegressionModel loadedModel = null;
+  private HashingTF tf = null;
+
+  public LogisticRegressionClassifier(Config config) {
+    this.config = config;
+  }
+
+  /**
+   * Configuration for the LogisticRegressionClassifier.
+   */
+  public static class Config extends MLPredictorConfig {
+    @Nullable
+    @Description("The number of features to use in training the model. It must be of type integer and equal to the" +
+                  " number of features used in LogisticRegressionTrainer. The default value if none is provided " +
+                  " will be 100.")
```

Comment:
nit: To avoid double spaces or missing spaces, we usually have the space at the end of the previous line, and don't start with a space at the start of the line. This, for example, will end up with 2 spaces.